src/bika/lims/content/worksheet.py

```diff
@@ -440,7 +440,7 @@ def nextRefAnalysesGroupID(self, reference):
             prefix = reference.id + "-D"
         bac = getToolByName(reference, 'senaite_catalog_analysis')
         ids = bac.Indexes['getReferenceAnalysesGroupID'].uniqueValues()
-        rr = re.compile("^" + prefix + "[\d+]+$")
+        rr = re.compile("^" + prefix + "[\\d+]+$")
```

Comment:
see https://docs.python.org/2/library/re.html

> each backslash must be expressed as \\ inside a regular Python string literal.

The new expression is functionally identical to the previous one. You can check yourself with the following code snippet:

```python
import re

print re.compile("\d").pattern == re.compile("\\d").pattern
```
(https://onecompiler.com/python2/3yf7b7tmk)

Using a single backslash works in this case but leads to a warning. It will fail if you actually want to match a backslash:

```python
import re

s = "abc \ xyz"
for p in [
  "^.+ \ .+$",
  "^.+ \\ .+$",
  "^.+ \\\ .+$",
  "^.+ \\\\ .+$",
  r"^.+ \ .+$",
  r"^.+ \\ .+$"
]:
  print "%s:\t%s" % (p, bool(re.compile(p).match(s)))
```

(https://onecompiler.com/python2/3yf7bgdzm)

Reply:
I was not aware of this, thanks for your explanation and code snippets!